src/Http/Controllers/WebhookController.php

```diff
@@ -54,12 +54,16 @@ public function stockPosition(Request $request)
             $stockPosition = $stockPosition[0];
             $daalderProduct = $this->productRepository->getProductFromExactId($stockPosition->ItemId);
 
-            $this->productRepository->storeStock($daalderProduct,[
-                'product_id' => $daalderProduct->id,
-                'in_stock' => $stockPosition->InStock ?? 0,
-                'planned_in' => $stockPosition->PlanningIn ?? 0,
-                'planned_out' => $stockPosition->PlanningOut ?? 0
-            ]);
+            if($daalderProduct) {
+                $this->productRepository->storeStock($daalderProduct,[
+                    'product_id' => $daalderProduct->id,
+                    'in_stock' => $stockPosition->InStock ?? 0,
+                    'planned_in' => $stockPosition->PlanningIn ?? 0,
+                    'planned_out' => $stockPosition->PlanningOut ?? 0
+                ]);
+            } else {
+                return response('', 422);
```

Comment:
Unfortunately, this may result in the webhook deregistering. Exact expects a 200, and assuming what they're sending is correct we may return a 200, so you can remove this else statement.

Reply:
Based on the logs, it gets  not found most of the time , including the test environment for which we tested this code, and the webhook is still there. So, what do you think?